Apps/WebClient/src/Server/Controllers/UserProfileController.cs

```diff
@@ -331,7 +339,9 @@ public async Task<IActionResult> UpdateUserEmail(string hdid, [FromBody] string
                 .Referer
                 .GetLeftPart(UriPartial.Authority);
 
-            bool result = this.userEmailService.UpdateUserEmail(hdid, emailAddress, new Uri(referer));
+            string bearerToken = this.httpContextAccessor.HttpContext.Request.Headers["Authorization"][0].Split(" ")[1];
```

Comment:
This seems more succinct:
` string accessToken = await this.httpContextAccessor.HttpContext.GetTokenAsync("access_token").ConfigureAwait(true);`

Its also consistent with what I did in lab.

Reply:
Yes, but is not testable since GetTokenAsync is an extension method :(

Reply:
Ahh but it is... See Laboratory Unit tests, pretty straight forward.

Reply:
uhm, the amount of stuff mocked for this is quite large...

Reply:
fixed